### PR TITLE
[Glimmer2] tests for html-safe and render

### PR DESCRIPTION
**Pull request by chadhietala**
/cc @krisselden @rwjblue 

Pretty sure we want to just use the SafeString from HandleBars, however it is unclear how we should import it sense it was a proxy import in HTMLBars.

### CODE REVIEW

**Comment by rwjblue** [comment]
@chadhietala - This is good to me.  Can you make https://github.com/emberjs/ember.js/blob/master/packages/ember-htmlbars/lib/utils/string.js a symlink to the `ember-glimmer` one, and expose it to `Ember.Handlebars` and `Ember.HTMLBars` in `ember-templates`?

File: packages/ember-glimmer/lib/utils/string.js [hunk 1]
```diff
@@ -0,0 +1 @@
+../../../ember-htmlbars/lib/utils/string.js
```
Comment (rwjblue) [inline packages/ember-glimmer/lib/utils/string.js]: I think this should go the other way since in the long term we are going to `rm -rf packages/ember-htmlbars`.

**Comment by rwjblue** [comment]
Overall this look good (one comment/tweak), can you confirm that `escapeExpression` is exposed by `glimmer`'s build?

**Comment by chadhietala** [comment]
What I ended up doing is moving the string module into `ember-glimmer` and also inlining the `SafeString` class and `escapExpression` functions as they came from HTMLBars. `SafeString` in HTMLBars is just a proxy to Handlebars.

File: packages/ember-templates/lib/index.js [hunk 2]
```diff
@@ -21,6 +29,39 @@ Ember.TextField = TextField;
 Ember.TextArea = TextArea;
 Ember.LinkComponent = LinkComponent;
 
+if (Ember.HTMLBars) {
+  Ember.HTMLBars.SafeString = SafeString;
+} else {
+  Ember.HTMLBars = { SafeString };
+}
+
+if (Ember.HandlerBars) {
```
Comment (rwjblue) [inline packages/ember-templates/lib/index.js]: Should be `Ember.Handlebars` (a few places below are typo'ed too).

File: packages/ember-templates/lib/index.js [hunk 3]
```diff
@@ -21,6 +29,39 @@ Ember.TextField = TextField;
 Ember.TextArea = TextArea;
 Ember.LinkComponent = LinkComponent;
 
+if (Ember.HTMLBars) {
+  Ember.HTMLBars.SafeString = SafeString;
```
Comment (rwjblue) [inline packages/ember-templates/lib/index.js]: Can you move these `Ember.HTMLBars` exports to `ember-templates/compat` (we already normalize and ensure `Ember.HTMLBars` is present there)?

File: packages/ember-templates/lib/index.js (hunk 2 above)
Comment (rwjblue) [inline packages/ember-templates/lib/index.js]: Can you also move these `Ember.Handlebars` exports to `ember-templates/compat`?

File: packages/ember-templates/lib/index.js [hunk 4]
```diff
@@ -21,6 +29,39 @@ Ember.TextField = TextField;
...
+    Ember.HandlerBars.Utils.escapeExpression = escapeExpression;
+  } else {
+    Ember.HandlerBars.Utils = { escapeExpression };
+  }
+} else {
+  Ember.HandlerBars = { SafeString, Utils: { escapeExpression } };
+}
+
+if (Ember.String) {
+  Ember.String.Utils = { htmlSafe };
```
Comment (rwjblue) [inline packages/ember-templates/lib/index.js]: This should be `Ember.String.htmlSafe` (no `Utils`).  Can you also import `String` from `ember-runtime`?

File: packages/ember-templates/lib/index.js [hunk 5]
```diff
@@ -21,6 +29,39 @@ Ember.TextField = TextField;
...
+} else {
```
Comment (rwjblue) [inline packages/ember-templates/lib/index.js]: No need to guard here, we should import `String`

File: packages/ember-views/lib/mixins/action_support.js [hunk 6]
```diff
@@ -132,7 +132,9 @@ export default Mixin.create({
       if (!shouldBubble) { return; }
     }
 
-    if (target = get(this, 'target')) {
+    target = get(this, 'target') || get(this, '_targetObject');
```
Comment (rwjblue) [inline packages/ember-views/lib/mixins/action_support.js]: Is this change related to the others?

Comment (chadhietala) [inline packages/ember-views/lib/mixins/action_support.js]: Not really I can PR separately. Just trying to reduce CI cycles 😁

**Comment by chadhietala** [comment]
Had to do some reorg and deletion of some similar work. Also moved the action test to #13850.

File: packages/ember-templates/lib/compat.js [hunk 7]
```diff
@@ -1,7 +1,46 @@
...
+      return SafeString;
+    }
+  });
+} else {
+  EmberHandlebars.SafeString = SafeString;
+}
+
+EmberHTMLBars.SafeString = SafeString;
 EmberHTMLBars.template = EmberHandlebars.template = template;
+EmberHTMLBarsUtils.escapeExpression = EmberHandleBarsUtils.escapeExpression = escapeExpression;
```
Comment (rwjblue) [inline packages/ember-templates/lib/compat.js]: I do not believe that we ever exposed this previously on `Ember.HTMLBars`, can you confirm and if we haven't fixup?

Comment (chadhietala) [inline packages/ember-templates/lib/compat.js]: Fixed.

**Comment by rwjblue** [comment]
LGTM
